src/infrastructure/content/server/ContentServer.java

```diff
@@ -0,0 +1,243 @@
+	 * This method is internally divided into 3 categories based on the meta key value
+	 * from the JSON object derived from argument message. If the meta value is newUser
+	 * then imageMap is updated with new userName and userImage and then the map is
+	 * stored in a JSON Array and sent to all active clients. If the meta value is 
+	 * message then current time is added to it and send to all active clients. If 
+	 * the meta value is userExit then the corresponding userImage from imageMap of
+	 * the userName which is exiting from is removed and informed to all other clients.
+	 * @param message - A String which actually represents a JSON object
+	 */
+	@Override
+	public void onMessageReceived(String message) {
+		// logging to logger
+		logMessage = "onMessageReceived method of ContentServer class is going to be executed";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+		// storing message in the form of JSON Object
+		JSONObject jsonObject;
+		try {
+			jsonObject = new JSONObject(message);
+		} catch(Exception e) {
+			// logging to logger
+			logMessage = "the argument message cannot be parsed as a JSON Object";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		try {
+			meta = jsonObject.getString("meta");
+		} catch(Exception e) {
+			// logging to logger
+			logMessage = "the value of meta key in argument message is not a String";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		if (meta.equals("newUser")) {
+			// logging to logger
+			logMessage = "New User case";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+			try {
+				userName = jsonObject.getString("username");
+			} catch(Exception e) {
+				// logging to logger
+				logMessage = "the value of username key in argument message is not a String";
+				logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+				return;
+			}
+			try {
+				userImage = jsonObject.getString("image");
+			} catch(Exception e) {
+				// logging to logger
+				logMessage = "the value of image key in argument message is not a String";
+				logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+				return;
+			}
+			// updating imageMap
+			imageMap.put(userName, userImage);
+			// declaring a JSON Array and a temporary JSON Object
+			JSONArray jsonArray = new JSONArray();
+			JSONObject tempJsonObject = new JSONObject();
+			// This loop adds all userName-userImage as a JSON object to a JSON Array
+			for (String name : imageMap.keySet()) {
+				userName = name;
+				userImage = imageMap.get(userName);
+				tempJsonObject.put("username", userName);
+				tempJsonObject.put("image", userImage);
+				jsonArray.put(tempJsonObject);
+				tempJsonObject.remove("username");
+				tempJsonObject.remove("image");
+			}
+			// logging to logger
+			logMessage = "The JSON Array is created which contains the userName to userImage Map";

```

Comment:
meta and username field from original JSON message needs to be added to new JSON Array. These values are expected in content's handler.

Reply:
Probably the JSON array is added to another JSON object and then the other fields are added